File: app/src/main/java/com/test/urovopaymentapp/utils/ClientException.java
package com.test.urovopaymentapp.utils;
/**Customized exception, used to save error code
 *
 */
public class ClientException extends Exception{
	private static final long serialVersionUID = -2155382375819767054L;

	private String errorCode;

	public ClientException(String errorCode) {
		super(errorCode);
		this.errorCode = errorCode;
	}

	public ClientException(Throwable cause, String errorCode) {
		super(cause);
		this.errorCode = errorCode;
	}

	/**Get exception code
	 * @return the errorCode
	 */
	public String getErrorCode() {
		return errorCode;
	}
}

File: app/src/main/java/com/test/urovopaymentapp/utils/exception/ClientException.kt
package com.test.urovopaymentapp.utils.exception

/**Customized exception, used to save error code
 *
 */
class ClientException : Exception {
    /**Get exception code
     * @return the errorCode
     */
    var errorCode: String
        private set

    constructor(errorCode: String) : super(errorCode) {
        this.errorCode = errorCode
    }

    constructor(cause: Throwable?, errorCode: String) : super(cause) {
        this.errorCode = errorCode
    }

    companion object {
        private const val serialVersionUID = -2155382375819767054L
    }
}
